Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx

```diff
@@ -117,47 +118,79 @@ const SubmissionDetails = () => {
+                    <CodeEditor
+                      readOnly
+                      code={currentSubmission?.content}
+                      selectedSubmissionType={submissionType}
+                    />
+                </div>
+                <div className={submissionDetailsClassName}>
+                    <Heading type={HeadingType.secondary}>{detailsHeadingText}</Heading>
+                    {isNil(currentSubmission)
+                        ? ''
+                        : (
+                            <SubmissionResults
+                              testRuns={currentSubmission.testRuns}
+                              compilerComment={currentSubmission?.compilerComment}
+                              isCompiledSuccessfully={currentSubmission?.isCompiledSuccessfully}
+                            />
+                        )}
 
-    return (
-        <div className={styles.detailsWrapper}>
-            <div className={styles.navigation}>
-                <div className={submissionsNavigationClassName}>
-                    <Heading type={HeadingType.secondary}>Submissions</Heading>
                 </div>
-                <RefreshableSubmissionsList
-                  items={currentProblemSubmissionResults}
-                  selectedSubmission={currentSubmission}
-                  className={styles.submissionsList}
-                />
-                { renderRetestButton() }
-                { renderSubmissionInfo() }
-            </div>
-            <div className={styles.code}>
-                <Heading
-                  type={HeadingType.secondary}
-                  className={styles.taskHeading}
-                >
-                    {problemNameHeadingText}
-                </Heading>
-                <CodeEditor
-                  readOnly
-                  code={currentSubmission?.content}
-                  selectedSubmissionType={submissionType}
-                />
-            </div>
-            <div className={submissionDetailsClassName}>
-                <Heading type={HeadingType.secondary}>{detailsHeadingText}</Heading>
-                <SubmissionResults
-                  testRuns={currentSubmission.testRuns}
-                  compilerComment={currentSubmission?.compilerComment}
-                  isCompiledSuccessfully={currentSubmission?.isCompiledSuccessfully}
-                />
             </div>
+        ),
+        [ currentProblemSubmissionResults,
+            currentSubmission,
+            detailsHeadingText,
+            problemNameHeadingText,
+            renderRetestButton,
+            renderSubmissionInfo,
+            submissionDetailsClassName,
+            submissionType ],
+    );
+
+    const renderErrorMessage = useCallback(() => (
+        <div className={styles.errorMessage}>
+            {validationResult.message}
         </div>
+    ), [ validationResult ]);
+
+    const renderPage = useCallback(
+        () => isNil(validationResult)
+            ? <div>Loading data</div>
```

Comment:
doesn't seem correct to check validation result is nil in the render page function

Reply:
That is how we did it when we implemented the contest validation and that is why I copied the same approach. I can try and think of something else if you think it is definitely wrong.

Reply:
agreed